src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -6,19 +6,121 @@
  * За бажанням можна реалізувати інші методи інтрефейсу Map.</p>
  */
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int capacity;
+    private int size;
+    private int threshold;
+    private Entry<K, V>[] table;
+
+    private static class Entry<K,V> {
+        final K key;
+        V value;
+        Entry<K,V> next;
+
+        Entry(K key, V value, Entry<K,V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    public MyHashMap() {
+        size = 0;
+        capacity = DEFAULT_INITIAL_CAPACITY;
+        table = new Entry[capacity];
+        threshold = countThreshold(capacity);
+    }
+
+    private int hash(K key) {
+        return  Math.abs(key.hashCode()) % table.length;
+    }
+
+    private int countThreshold(int capacity) {
+        return Math.round(capacity * DEFAULT_LOAD_FACTOR);
+    }
+
+    private Entry<K,V> checkKey(K key) {
+        if (key == null) {
+            Entry<K,V> bucket = table[0];
+            while (bucket != null) {
+                if (bucket.next == null && bucket.key == null) {
+                    return bucket;
+                }
+                bucket = bucket.next;
+            }
+        } else {
+            int index = hash(key);
+            Entry<K, V> bucket = table[index];
+            while (bucket != null) {
+                if (bucket.key.equals(key)) {
+                    return bucket;
+                }
+                bucket = bucket.next;
+            }
+        }
+        return null;
+    }
+
+    private void addNewEntry(K key, V value) {
+        int index = hash(key);
+        Entry<K,V> bucket = table[index];
+        table[index] = new Entry(key, value, bucket);
+        size++;
+    }
+
+    private void putForNullKey(V value) {
+        if (checkKey(null) == null) {
```

Comment:
better write result `checkKey(null)` in the separate method

Reply:
@Sofasmile I did not understand  :( .Do I need to split the method  checkKey into two separate methods checkKey and checkKeyNull?

Reply:
I will add a  variable to write the result of the checkKey method.